Mathlib/RingTheory/Polynomial/SmallDegreeVieta.lean

```diff
@@ -15,6 +16,10 @@ namespace Polynomial
 
 variable {R T S : Type*}
 
+lemma quadratic_ne_zero [Semiring R] {a b c : R} (ha : a ≠ 0) : C a * X ^ 2 + C b * X + C c ≠ 0 :=
+  fun hx ↦ ha (by rw [show a = (C a * X ^ 2 + C b * X + C c).coeff 2 by
+    simp, hx, coeff_zero])
```

Comment:
I don't think this deserves to be a lemma. It is only a fragment of the contrapositive of the following useful lemma:
```suggestion
@[simp]
lemma quadratic_eq_zero_iff [Semiring R] {a b c : R} :
    C a * X ^ 2 + C b * X + C c = 0 ↔ a = 0 ∧ b = 0 ∧ c = 0 := by
  constructor
  · rw [Polynomial.ext_iff]
    exact fun h ↦ ⟨by simpa using h 2, by simpa using h 1, by simpa using h 0⟩
  · rintro ⟨rfl, rfl, rfl⟩
    simp
```
However what would be really desirable here is a `simproc` which worked for any degree.

Reply:
How about:
```
lemma zero_iff [Semiring R] {p : R[X]} : p = 0 ↔ ∀ (n : ℕ), coeff p n = 0 := ext_iff

lemma not_zero_iff [Semiring R] {p : R[X]} : ¬ p = 0 ↔ ∃ (n : ℕ), ¬ coeff p n = 0 := by
  rw [zero_iff, not_forall]
```
which can then be invoked with `not_zero_iff.mpr ⟨2, by simp [ha]⟩` to get the desired goal.